### PR TITLE
upload all changues

### PR DESCRIPTION
**Pull request by bernatbrunet**
Created APIs for managing order, order_line, and customer entities.
	•	Developed an additional API for setting up test databases specifically used for testing purposes.
	•	Removed db_insert and replaced it with db_insert_test for testing workflows.
	•	Created and successfully tested automated tests for all the newly developed APIs.

### CODE REVIEW

File: api/tenantdb.js [hunk 1]
```diff
@@ -19,24 +20,45 @@ module.exports = {
...
+            user: db.user,
+            password: db.password,
+            database: db.database,
             connectionLimit: db.connectionLimit
         });
         const promisePool = pool.promise();
         connectionMap.set(db.id, { pool, promisePool });
     },
 
+    // Add connection explicitly for tenant 999
```
Comment (jordidh) [inline api/tenantdb.js]: This connection must use environtment variables

File: estructura_proyecto.txt
Comment (jordidh) [inline estructura_proyecto.txt:1]: What is this file?

File: structure.txt
Comment (jordidh) [inline structure.txt:1]: What is this file?

File: api/tenantdb.js [hunk 2]
```diff
@@ -19,24 +20,45 @@ module.exports = {
...
 
+    // Add a new connection manually
     addConnection: function (db) {
         const pool = mysql.createPool({
-            host: db.db_host,
-            port: db.db_port,
-            user: db.db_username,
-            password: db.db_password,
-            database: db.db_name,
+            host: db.host,
```
Comment (jordidh) [inline api/tenantdb.js]: This environment variables must be specified in the env.example file

File: app.js [hunk 3]
```diff
@@ -1,262 +1,98 @@
...
     },
-    apis: ['./routes/*.js'] // Verifica que aquesta línia inclou el camí correcte
+    apis: ['./routes/v1/*.js'], // Path to the API docs in the routes
 };
 
-const swaggerDocs = swaggerJsDoc(swaggerOptions);
+const swaggerDocs = swaggerJsdoc(swaggerOptions);
 app.use('/v1/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs));
 
-// Gets all databases from mysql and adds to the connectionMap with tenantdb.connectAll(dbs)
```
Comment (jordidh) [inline app.js:93]: Where is this function?

File: routes/clean-db-test.js [hunk 4]
```diff
@@ -32,37 +32,37 @@ router.get(('/'), async function (req, res, next) {
         (1, 'UNIT01', 'descripcio de prova1', 1), 
         (2, 'UNIT02', 'descripcio de prova', 10);`);
         if (insertUnit.length !== 2) {
-            throw new Error('Couldn\'t insert values into db_test.unit');
+            throw new Error('Couldn\'t insert values into db_test_insert.unit');
```
Comment (jordidh) [inline routes/clean-db-test.js]: Must show databasename from environment variables

File: test/routes/v1/1_warehouse_spec.js [hunk 5]
```diff
@@ -41,90 +42,132 @@ const UNIT10 = {
...
-        const locationLimits = await chai.request(URL).get('/location' + idTenantProva);
-        expect(locationLimits.statusCode).to.equal(200);
-        expect(locationLimits.body.data).to.be.an('array');
-        expect(locationLimits.body.errors).to.be.an('array');
-        expect(locationLimits.body.errors).to.be.an('array').that.eql([]);
+        const locations = await chai.request(URL).get(`/location?id=${TENANT_ID_TEST}`);
+        expect(locations.statusCode).to.equal(200);
+        expect(locations.body.data).to.be.an('array');
+        
+        if (locations.body.requestId) {
```
Comment (jordidh) [inline test/routes/v1/1_warehouse_spec.js]: Why this if?

File: test/routes/v1/2_order_spec.js [hunk 6]
```diff
@@ -23,52 +24,64 @@ const ORDER_NEW = {
...
-        console.log(orders.body);
-        expect(orders.status).to.equal(200);
-        expect(orders.body.requestId).to.be.a('string');
+        const orders = await chai.request(URL).get(`?id=${TENANT_ID_TEST}`);
+        expect(orders.statusCode).to.equal(200);
         expect(orders.body.data).to.be.an('array');
-        expect(orders.body.errors).to.be.an('array').that.is.empty;
+        
+        // Verifica si 'requestId' está presente antes de hacer la comparación
+        if (orders.body.requestId) {
```
Comment (jordidh) [inline test/routes/v1/2_order_spec.js]: Why this if?

File: test/routes/v1/4_customer_spec.js [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+    it('should create a new customer', async () => {
+        try {
+            console.log('Creating a new customer...');
+            const res = await chai.request(URL).post('/' + idTenantProva).send(CUSTOMER_NEW); // Send POST request to create customer
+            console.log('Create customer response:', res.body);
+            expect(res.statusCode).to.equal(201); // Expect successful creation status
+            expect(res.body.data).to.be.an('object'); // Response should contain a customer object
+            expect(res.body.data.id).to.exist; // Customer ID should be present
+            if (res.body.requestId) {
```
Comment (jordidh) [inline test/routes/v1/4_customer_spec.js]: Why this if?

**Review by jordidh — CHANGES_REQUESTED** [review]
There are some issues that has to be fixed

File: app.js (hunk 3 above)
Comment (bernatbrunet) [reply app.js:93]: The connectAll function is located in the api/tenantdb.js file.

File: api/tenantdb.js (hunk 2 above)
Comment (bernatbrunet) [reply api/tenantdb.js]: Problem solved.

File: structure.txt
Comment (bernatbrunet) [reply structure.txt:1]: Problem solved.

File: estructura_proyecto.txt
Comment (bernatbrunet) [reply estructura_proyecto.txt:1]: Problem solved.

File: api/tenantdb.js (hunk 1 above)
Comment (bernatbrunet) [reply api/tenantdb.js]: Problem solved.

File: routes/clean-db-test.js (hunk 4 above)
Comment (bernatbrunet) [reply routes/clean-db-test.js]: Problem solved.

File: test/routes/v1/1_warehouse_spec.js (hunk 5 above)
Comment (bernatbrunet) [reply test/routes/v1/1_warehouse_spec.js]: Problem solved.

File: test/routes/v1/2_order_spec.js (hunk 6 above)
Comment (bernatbrunet) [reply test/routes/v1/2_order_spec.js]: Problem solved.

File: test/routes/v1/4_customer_spec.js (hunk 7 above)
Comment (bernatbrunet) [reply test/routes/v1/4_customer_spec.js]: Problem solved.

File: app.js [hunk 8]
```diff
@@ -1,262 +1,171 @@
...
-            }
-            const db = {
+            const dbConfig = {
                 id: tenant.id,
-                host: process.env.DB_HOST,
-                port: process.env.DB_PORT,
-                user: resultQuery[0][0].db_username,
-                password: resultQuery[0][0].db_password,
-                database: resultQuery[0][0].db_name,
+                host: tenant.db_host || process.env.DB_HOST,
```
Comment (jordidh) [inline app.js]: Delete environment variables

**Review by jordidh — CHANGES_REQUESTED** [review]

**Review by jordidh — APPROVED** [review]

**Pull request merged**